### PR TITLE
Replace `push_back` to `emplace_back` and code refactor

### PR DESCRIPTION
**Pull request by GermanAizek**
@Xottab-DUTY, code refactor with R2 sun and renderer GL.

### CODE REVIEW

**Comment by GermanAizek** [comment]
start
@Xottab-DUTY

File: src/Layers/xrRenderPC_GL/r2_R_sun.cpp [hunk 1]
```diff
@@ -18,9 +18,6 @@ const float tweak_guaranteed_range = 20.f; //. ?
 // float			OLES_SUN_LIMIT_27_01_07			= 180.f		;
 float OLES_SUN_LIMIT_27_01_07 = 100.f;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderPC_GL/r2_R_sun.cpp]:
```suggestion

const float MAP_SIZE_START = 6.f;
const float MAP_GROW_FACTOR = 4.f;

```

File: src/xrCore/FS.h [hunk 2]
```diff
@@ -408,7 +408,7 @@ class XRCORE_API IReader : public IReaderBase<IReader>
     // iterators
     IReader* open_chunk_iterator(u32& ID, IReader* previous = nullptr); // NULL=first
 
-    size_t find_chunk(u32 ID, bool* bCompressed = 0);
+    virtual size_t find_chunk(u32 ID, bool* bCompressed = 0);
```
Comment (Xottab-DUTY) [inline src/xrCore/FS.h]:
```suggestion
    size_t find_chunk(u32 ID, bool* bCompressed = 0);
```

**Comment by Xottab-DUTY** [comment]
> start @Xottab-DUTY

What is this image about?

File: src/xrCore/Threading/TaskManager.hpp [hunk 3]
```diff
@@ -56,12 +56,12 @@ class XRCORE_API TaskManager final
 public:
     // TaskFunc is at the end for fancy in-place lambdas
     // Create a task, but don't run it yet
-    [[nodiscard]] Task& CreateTask(pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
-    [[nodiscard]] Task& CreateTask(pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
+    [[nodiscard]] static Task& CreateTask(pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
+    [[nodiscard]] static Task& CreateTask(pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
```
Comment (Xottab-DUTY) [inline src/xrCore/Threading/TaskManager.hpp]:
```suggestion
    [[nodiscard]] Task& CreateTask(pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
    [[nodiscard]] Task& CreateTask(pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
```

File: src/xrCore/Threading/TaskManager.hpp [hunk 4]
```diff
@@ -56,12 +56,12 @@ class XRCORE_API TaskManager final
...
 
     // Create a task as child, but don't run it yet
-    [[nodiscard]] Task& CreateTask(Task& parent, pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
-    [[nodiscard]] Task& CreateTask(Task& parent, pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
+    [[nodiscard]] static Task& CreateTask(Task& parent, pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
+    [[nodiscard]] static Task& CreateTask(Task& parent, pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
```
Comment (Xottab-DUTY) [inline src/xrCore/Threading/TaskManager.hpp]:
```suggestion
    [[nodiscard]] Task& CreateTask(Task& parent, pcstr name, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
    [[nodiscard]] Task& CreateTask(Task& parent, pcstr name, const Task::OnFinishFunc& onFinishCallback, const Task::TaskFunc& taskFunc, size_t dataSize = 0, void* data = nullptr);
```

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**